ambry-api/src/main/java/com/github/ambry/account/Container.java

```diff
@@ -125,6 +125,12 @@ public class Container {
    */
   public static final String DEFAULT_PRIVATE_CONTAINER_NAME = "default-private-container";
 
+  /**
+   * Default name for the containers associated with S3 APIs. Since, S3 requests on client side only take Account
+   * (i.e. Bucket) name, we use a default name for containers.
+   */
+  public static final String DEFAULT_S3_CONTAINER_NAME = "container-a";
```

Comment:
Is this our final solution or temporary solution?
I thought we would go with
1. The endpoint is
https://ambry-frontend-main.ambry-frontend.xxxx.linkedin.com:3183/s3
2. The S3 path is
```
"s3://<ambry-account-name>/<ambry-container-name>/<folder-name>/<file-name>"
```

Reply:
This was actually @justinlin-linkedin's idea in the doc.  Since S3 has the concept of only buckets, I am also wondering that some S3 APIs may assume that path after `/bucket-name/` is the `key-name`.  For example, they have `HEAD /bucket-name` API. 

We may have to also tell S3 customers that they can have only 1 container (with name `container-a`) under Ambry accounts. 

What do you think?

Reply:
Sounds good to me so far.
Let's discuss it with our customers next time including the ttl and path.

Reply:
Hmm, I am thinking that we shouldn't have any scalability concerns. We use container as a logically group blobs with similar properties (encrypted, cacheable, media-scan). 

Containers also provide ability to compact all blobs in it in background. It may happen that compaction of containers may take longer time. But I am thinking that it should be okay as it is asynchronous process and no one is actually waiting on it.

Reply:
Got it, this makes sense. I think then we could use different account for different quality of service.

Reply:
Do we want to use a different container name for default s3 container? container-a is too common amount accounts. If customers want to enable s3 API for its application, it should put files under s3 specific container. Something like "s3" might be better.